src/states/state-handlers.ts

```diff
@@ -228,6 +239,46 @@ export const handleLoggedIn = async (ctx: BotHandlerContext): Promise<User> => {
   return updatedUser;
 };
 
+export const handleReplyAttendanceButton = async (ctx: BotHandlerContext): Promise<User> => {
+  const { payload } = ctx;
+  const updatedUser = structuredClone(ctx.user);
+
+  // Check if the user clicked either "Today's Attendance" or "Yesterday's Attendance"
+  if (payload.button.text === "Today's Attendance" || payload.button.text === "Yesterday's Attendance") {
+    const today = new Date();
+    let selectedDate;
+
+    if (payload.button.text === "Today's Attendance") {
+      // User clicked "Today's Attendance"
+      selectedDate = today;
+    } else {
+      // User clicked "Yesterday's Attendance"
+      const yesterday = new Date(today);
+      yesterday.setDate(today.getDate() - 1);
```

Comment:
This will break on the 1st of every month. Instead, manipulate the numeric timestamp representation of the date and convert it back to a date object. Already being done in the date list for your reference:
https://github.com/asetalias/amibot/blob/main/src/states/render-messages.ts#L26-L28